cmd/ci-secret-generator/main.go

```diff
@@ -0,0 +1,226 @@
+	fs.StringVar(&o.logLevel, "log-level", "info", fmt.Sprintf("Log level is one of %v.", logrus.AllLevels))
+	fs.IntVar(&o.maxConcurrency, "concurrency", 1, "Maximum number of concurrent in-flight goroutines to BitWarden.")
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Errorf("cannot parse args: %q", os.Args[1:])
+	}
+	return o
+}
+
+func (o *options) validateOptions() error {
+	level, err := logrus.ParseLevel(o.logLevel)
+	if err != nil {
+		return fmt.Errorf("invalid log level specified: %w", err)
+	}
+	logrus.SetLevel(level)
+	if o.bwUser == "" {
+		return fmt.Errorf("--bw-user is empty")
+	}
+	if o.bwPasswordPath == "" {
+		return fmt.Errorf("--bw-password-path is empty")
+	}
+	if o.configPath == "" {
+		return fmt.Errorf("--config is empty")
+	}
+	return nil
+}
+
+func (o *options) completeOptions(secrets sets.String) error {
+	bytes, err := ioutil.ReadFile(o.bwPasswordPath)
+	if err != nil {
+		return err
+	}
+	o.bwPassword = strings.TrimSpace(string(bytes))
+	secrets.Insert(o.bwPassword)
+
+	bytes, err = ioutil.ReadFile(o.configPath)
+	if err != nil {
+		return err
+	}
+
+	err = yaml.Unmarshal(bytes, &o.config)
+	if err != nil {
+		return err
+	}
+	return o.validateCompletedOptions()
+}
+
+func (o *options) validateCompletedOptions() error {
+	if o.bwPassword == "" {
+		return fmt.Errorf("--bw-password-file was empty")
+	}
+
+	for i, bwItem := range o.config {
+		if bwItem.ItemName == "" {
+			return fmt.Errorf("config[%d].itemName: empty key is not allowed", i)
+		}
+		switch bwItem.Attribute.Name {
+		case attributeTypePassword, "":
+		default:
+			return fmt.Errorf("config[%d].attribute: only the '%s' is supported, not %s", i, attributeTypePassword, bwItem.Attribute.Name)
+		}
+		if (bwItem.Field.Name != "" && bwItem.Field.Cmd == "") ||
+			(bwItem.Attribute.Name != "" && bwItem.Attribute.Cmd == "") ||
+			(bwItem.Attachment.Name != "" && bwItem.Attachment.Cmd == "") {
+			return fmt.Errorf("config[%d]: empty field not allowed for cmd if name is specified for any of attribute, field or attachment", i)
+		}
+	}
+	return nil
+}
+
+func executeCommand(command string) ([]byte, error) {
+	cmd := strings.Fields(command)
+	out, err := exec.Command(cmd[0], cmd[1:]...).CombinedOutput()
+	if err != nil {
+		return nil, fmt.Errorf("bw cmd failed: %v", string(out))
+	}
+	return out, nil
+}
+
+func processItem(w io.Writer, cmd string, entryType string, dryRun bool, procesor func([]byte) error) error {
+	out, err := executeCommand(cmd)
+	if err != nil {
+		return fmt.Errorf("failed to execute command: %w", err)
+	}
+	if dryRun {
+		if _, err := fmt.Fprintf(w, "\t%s: %s", entryType, string(out)); err != nil {
+			return err
+		}
+	} else {
+		return procesor(out)
+	}
+	return nil
+}
+
+func updateSecrets(bwItems []bitWardenItem, bwClient bitwarden.Client, dryRun bool) error {
+	if dryRun {
+		logrus.Infof("Dry-Run enabled, writing secrets to stdout")
```

Comment:
Yeah I know you copied this elsewhere, but we can never write secrets to stdout, they must be written to a file. If ppl _really_ want that because they run this locally or sth, they can `cat` the file afterwards. Printing it introduces the possibility that someone accidentally does that in a job run